### DISCUSSION TITLE
Update branch triggered through API does not run checks

### DISCUSSION THREAD

**Discussion by patrickdet** [community/community#26520] [Actions]
<p>Hello everyone,</p>
<p>we have the following workflow which triggers the update branch api call (<a href="https://developer.github.com/v3/pulls/#update-a-pull-request-branch" rel="nofollow noopener">https://developer.github.com/v3/pulls/#update-a-pull-request-branch</a>) for every open PR as soon as our default branch (staging in our case), that all PRs are pointing to, has changed. The idea is that we want to save us the manual clicks on the “Update Branch” button for every PR.</p>
<p>The staging branch gets merged into each PR but the problem is that non of the checks (github actions workflows) are being executed afterwards. They are stuck in “Expected — Waiting for status to be reported” but they are never actually run. </p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/6/6c734559ed1885079f05e6dbda03aae4a4a6e943.png" data-download-href="/uploads/short-url/ftox554jiwR40rJCTSsD3YYycHV.png?dl=1" title="Screenshot 2020-04-15 at 10.31.39.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/2510/181098130-e3696451-a619-445b-a350-b9bb7d5485e2.png" alt="Screenshot 2020-04-15 at 10.31.39.png" data-base62-sha1="ftox554jiwR40rJCTSsD3YYycHV" width="690" height="219" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/6/6c734559ed1885079f05e6dbda03aae4a4a6e943_2_690x219.png, https://aws1.discourse-cdn.com/github/optimized/2X/6/6c734559ed1885079f05e6dbda03aae4a4a6e943_2_1035x328.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/6/6c734559ed1885079f05e6dbda03aae4a4a6e943_2_1380x438.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/6/6c734559ed1885079f05e6dbda03aae4a4a6e943_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screenshot 2020-04-15 at 10.31.39.png</span><span class="informations">1510×480 51.8 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>When I look at the “Checks” tab it is empty:</p>
<p><img src="https://user-images.githubusercontent.com/2510/181098131-858ea2e3-b8f9-4e59-8f4f-820cc7a2d1c6.png" alt="Screenshot 2020-04-15 at 10.31.49.png" data-base62-sha1="j3LMxyiWH0vPsx66DPP0PKOnEu5" width="240" height="86"></p>
<p>The checks are executed fine for subsequent commits on the PR’s branch. They are also executed when we click the “Update Branch” button from the UI. </p>
<p>Any input would be appreciated. </p>
<p>Workflow:</p>
<pre><code>on:
  push:
    branches:
      - staging

jobs:
  update-pr:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/github-script@0.9.0
        with:
          github-token: ${{secrets.GITHUB_TOKEN}}
          script: |
            const prsResponse = await github.pulls.list({
              owner: context.repo.owner,
              repo: context.repo.repo,
            });

            await Promise.all(prsResponse.data.map(async (pr) =&gt; {
              if (pr.base.ref !== 'staging') return Promise.resolve();

              return github.pulls.updateBranch({
                owner: context.repo.owner,
                repo: context.repo.repo,
                pull_number: pr.number,
              });
            }))</code></pre>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/patrickdet">@patrickdet</a> ,</p>
<p>I notice you are using the <strong>GITHUB_TOKEN</strong> to authenticate on the " <strong>update pull request branch</strong>" API.<br>
There is a note you need to know, you also can reference the docs about <a href="https://help.github.com/en/actions/configuring-and-managing-workflows/authenticating-with-the-github_token#using-the-github_token-in-a-workflow" rel="nofollow noopener"><strong>authenticating with the GITHUB_TOKEN</strong></a>:</p>
<blockquote>
<p>Events triggered by the GITHUB_TOKEN will not create a new workflow run, this prevents you from accidentally creating recursive workflow runs.</p>
</blockquote>
<p>If you really need the " <strong>update pull request branch</strong>" API also can trigger the workflow run, you can create a personal access token and set it as a secret in your repository, then use this personal access token to authenticate. But you also should avoid the recursive workflow runs.</p>

**Reply by matherm-aboehm** [reply]
Hi @BrightRan ,
does this also apply for the `createCommitOnBranch` GraphQL API?

I try to automate the task to append a fixup commit on a PR branch created by @dependabot , so that checks that first fails on the plain PR without the fixup commit, will then be re-checked again on the new commit and eventually succeed.
Currently for the signed commit I use the [planetscale/ghcommit-action](https://github.com/planetscale/ghcommit-action) action on a `workflow_run` trigger, which runs when the check run, that needs this fixup, fails.
Is there any way to do this without a PAT?

**Comment by breneckd** [comment]
What about tokens created by a github app?
